lib/main.dart

```diff
@@ -99,9 +98,9 @@ class _FilesHomeState extends State<FilesHome> {
     for (final MapEntry element in folderProvider.directories) {
       sideDestinations.add(
         SideDestination(
-          element.value,
-          Utils.getEntityName(element.key),
-          element.key,
+          element.value as IconData,
+          Utils.getEntityName(element.key as String),
+          element.key as String,
```

Comment:
for all the typecasts here it would be better to instead explicit the type on the `MapEntry element` above, like `MapEntry<String, IconData> element`

Reply:
Looks like `for (final element in folderProvider.directories)` we need not have explicit data type to be defined here. `folderProvider.directories` already has `MapEntry<String, IconData>` data type